dist/streams.js

```diff
@@ -55,8 +57,8 @@ class MetacomReadable extends EventEmitter {
       await this.waitEvent(PULL_EVENT);
       return this.push(data);
     }
-    this.queue.push(data);
-    if (this.queue.length === 1) this.emit(PUSH_EVENT);
+    if (data) this.queue.push(data);
```

Comment:
Please explain this change

Reply:
> Please explain this change

data can be null

Reply:
1. In this case you're skipping not only `null`, but all nullish values like `0` or `''`.
It's better to allow all possible values by replaceing `null` terminator with a symbol.
Consider to add these changes:

```js
const STREAM_ID_LENGTH = 4;
const STREAM_FINISHER = Symbol(); // Added symbol
```

```js
async stop() {
  if (this.packetsRead === this.expectedPackets) {
    this.streaming = false;
    this.emit(PUSH_EVENT, STREAM_FINISHER);
    return;
  }
  await this.waitEvent(PULL_EVENT);
  await this.stop();
}
```

```js
async read() {
  if (this.queue.length > 0) return this.pull();
  const finisher = await this.waitEvent(PUSH_EVENT);
  if (finisher === STREAM_FINISHER) return STREAM_FINISHER;
  return this.pull();
}
```

```js
async *[Symbol.asyncIterator]() {
  while (this.streaming) {
    const chunk = await this.read();
    if (chunk === STREAM_FINISHER) return;
    yield chunk;
  }
}
```

2. If you remove `if (this.queue.length === 1)`, then stream emits PUSH_EVENT on every push, which is redundant, because readable is being read while `this.queue` is not empty. Why it was removed?